source/common/http/conn_manager_impl.cc

```diff
@@ -437,6 +445,12 @@ void ConnectionManagerImpl::ActiveStream::onIdleTimeout() {
   }
 }
 
+void ConnectionManagerImpl::ActiveStream::onRequestTimeout() {
+  connection_manager_.stats_.named_.downstream_rq_timeout_.inc();
+  sendLocalReply(request_headers_ != nullptr && Grpc::Common::hasGrpcContentType(*request_headers_),
```

Comment:
What prevents a sendLocalReply() invoked from here from being followed by an actual response from upstream?   I.e. what guarantees the flip side of the above assert, which is that we don't send a global reply (from upstream) after we've sent a local reply?

Reply:
Your explanation covers the bases as far as I'm concerned; no need to look more closely at the flow on my behalf.